chemreg/compound/tests/test_views.py

```diff
@@ -171,3 +172,56 @@ def test_detail_view(self):
     def test_destroy_view(self):
         response = self.client.delete(self.detail_url)
         assert response.status_code == 204
+
+
+class TestQueryStructureTypeViewSet(APITestCase):
+    def setUp(self):
+        self.qst = QueryStructureTypeFactory()
+        self.qst2 = QueryStructureTypeFactory()
+        self.list_url = reverse("querystructuretype-list")
+        self.detail_url = reverse(
+            "querystructuretype-detail", kwargs={"name": self.qst.name}
+        )
+        self.paginated_url = f"{self.list_url}?size=5&page=2"
+
+    def test_create_view(self):
+        new = {
+            "name": "new-querystructuretype",
+            "label": "a new query structure type",
+            "short_description": "described here",
+            "long_description": "described at greater length here",
+        }
```

Comment:
Can we use `QueryStructureTypeJSONFactory` for these?

Reply:
Switched.